woocommerce/woocommerce
# Add redirect to new In-App marketplace installer

### Changes proposed in this Pull Request:

Closes https://github.com/Automattic/woocommerce.com/issues/18953

The new In-App Marketplace has an improved installer flow that is handled from the subscription page. The installer can be automatically triggered to install a product based on a product key provided in a URL parameter (`install=[PRODUCT_KEY]`). The key can be passed automatically in various places in a user journey and checkout flow to make the process of purchase and installation much more seamless.

This PR adds a check to the old marketplace page for this URL parameter, and will redirect to the new marketplace, preserving that URL parameter to be used in future development.

Note, we are applying this to the old marketplace to retain backwards compatibility. Redirecting users there means that users who are on older versions of WooCommerce before this change will simply see no redirect happen but still be able to install manually.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally
2. Visit the old Marketplace at http://localhost:8888/wp-admin/admin.php?page=wc-addons&section=_all
3. See that this still works without issues
4. Visit the old Marketplace using the following URL with the `install` parameter included — http://localhost:8888/wp-admin/admin.php?page=wc-addons&section=_all&install=W00-08fa9a3f-4847-a407-eb4aeb0f6e31
5. See that you are redirected to the new marketplace with the `install` parameter in-tact
6. Check that visiting a URL with the key in an unexpected format doesn't trigger a redirect — http://localhost:8888/wp-admin/admin.php?page=wc-addons&section=_all&install=12345
7. Test around this issue, particularly looking at any potential security concerns around the formatting and escaping.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Redirect users to new In-app marketplace installer if correct URL parameter is provided

#### Comment 

</details>
